src/platform/testing/e2e/cypress/support/form-tester/utilities.js

```diff
@@ -18,7 +18,7 @@ export const createArrayPageObjects = formConfig => {
 
   return (arrayPages || []).map(({ arrayPath, path }) => ({
     arrayPath,
-    regex: new RegExp(path.replace(':index', '(\\d+)$')),
+    regex: new RegExp(path.replace(':index', '(\\d+)')),
```

Comment:
this was a bug I found, where cypress wasn't happy with `/route/:index/something`. It wanted `:index` to be at the end `$` of the path, but it should instead be able to go anywhere in the string.